### PR TITLE
feat(cli/datacontract): Add data quality assertion support

### PR DESCRIPTION
**Pull request by asikowitz**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/api/entities/datacontract/datacontract.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ class DataContract(ConfigModel):
     freshness: Optional[FreshnessAssertion] = pydantic.Field(default=None)
 
     # TODO: Add a validator to ensure that ids are unique
-    data_quality: Optional[List[DataQualityAssertion]] = None
+    data_quality: Optional[List[DataQualityAssertion]] = pydantic.Field(default=None)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/api/entities/datacontract/datacontract.py:57]: interesting - these two are equivalent right?

File: metadata-ingestion/src/datahub/ingestion/api/incremental_lineage_helper.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.graph.client import DataHubGraph
+from datahub.metadata.schema_classes import (
+    FineGrainedLineageClass,
+    MetadataChangeEventClass,
+    SystemMetadataClass,
+    UpstreamClass,
+    UpstreamLineageClass,
+)
+from datahub.specific.dataset import DatasetPatchBuilder
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/api/incremental_lineage_helper.py:15]: looks like the base commit isn't set correct?

**Review by hsheth2 — APPROVED** [review]
broadly looks reasonable

**Comment by hsheth2** [comment]
@asikowitz btw we should probably write some tests for the contracts cli

**Comment by jjoyce0510** [comment]
What exactly is this contracts CLI going to be used for? 

Can we provide some examples? 

Also, we are moving quickly on new assertion types, how should we think about adding new types of assertions into contracts?

File: metadata-ingestion/src/datahub/api/entities/datacontract/data_quality_assertion.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
 from typing_extensions import Literal
 
 import datahub.emitter.mce_builder as builder
+from datahub.api.entities.datacontract.assertion import BaseAssertion
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/api/entities/datacontract/data_quality_assertion.py:7]: Should we also support Volume Assertions + Freshness Assertions? 

Or are these already supported

Comment (asikowitz) [reply metadata-ingestion/src/datahub/api/entities/datacontract/data_quality_assertion.py:7]: Freshness is supported I believe, or at least a subset of it. Volume not yet

**Comment by asikowitz** [comment]
https://linear.app/acryl-data/issue/ING-391/add-tests-for-data-contract-cli-support-more-assertions created

**Pull request merged**